src/vue-property-decorator.ts

```diff
@@ -13,10 +14,27 @@ export type Constructor = {
  * @param  options the option for the prop
  * @return PropertyDecorator
  */
-export function Prop(options: (PropOptions | Constructor | Constructor[]) = { type: null }): PropertyDecorator {
-  return createDecorator((componentOptions, key) => {
-      (componentOptions.props || (componentOptions.props = {}) as any)[key] = options
-    })
+export function Prop(target: Vue, key: string): void
+export function Prop(target: (PropOptions | Constructor[])): PropertyDecorator
+export function Prop(options: (Vue | PropOptions | Constructor[]), key?: string): void | PropertyDecorator {
```

Comment:
I have no idea how to make it work when adding in the single `Constructor`. Because I have no idea how to test whether the input was a single `Constructor` or an instance of `PropOptions`. So I'm guessing for now it's best to just let TypeScript think they are the same thing. If it ain't broke don't fix it?